spec/activerecord-bitemporal/transaction_at_spec.rb

```diff
@@ -33,4 +37,347 @@
+
+      shared_examples "invalid uniqueness" do
+        include_context "define active models"
+        subject { EmployeeWithUniquness.new(name: "Jane", created_at: new_from, deleted_at: new_to) }
+        it { is_expected.to be_invalid }
+      end
+
+      # active1 transaction time : |<---------->|
+      # active2 transaction time :                                 |<---------->|
+      # new transaction time     :                 |<---------->|
+      it_behaves_like "valid uniqueness" do
+        let(:new_from) { active1_to + 2.days }
+        let(:new_to) { new_from + 10.days }
+      end
+
+      # active1 transaction time :                 |<---------->|
+      # active2 transaction time :                                 |<---------->|
+      # new transaction time     : |<---------->|
+      it_behaves_like "valid uniqueness" do
+        let(:new_from) { new_to - 10.days }
+        let(:new_to) { active1_from - 2.days }
+      end
+
+      # active1 transaction time : |<---------->|
+      # active2 transaction time :                |<---------->|
+      # new transaction time     :                               |<---------->|
+      it_behaves_like "valid uniqueness" do
+        let(:new_from) { active2_to + 2.days }
+        let(:new_to) { new_from + 10.days }
+      end
+
+      # active1 transaction time : |<---------->|
+      # active2 transaction time :          |<---------->|
+      # new transaction time     :                    |<---------->|
+      it_behaves_like "invalid uniqueness" do
+        let(:new_from) { active1_to - 2.days }
+        let(:new_to) { active2_from + 2.days }
+      end
+
+      # active1 transaction time : |<---------->|
+      # active2 transaction time :              |<---------->|
+      # new transaction time     :                           |<---------->|
+      it_behaves_like "valid uniqueness" do
+        let(:new_from) { active2_to }
+        let(:new_to) { new_from + 10.days }
+      end
+    end
+
+    describe ".create" do
+      subject { -> { EmployeeWithUniquness.create!(name: "Tom") } }
+      context "exists destroyed model" do
+        let(:employee) { EmployeeWithUniquness.create!(name: "Jane").tap { |it| it.update(name: "Tom") } }
+        before do
+          employee.destroy!
+        end
+        it { is_expected.not_to raise_error }
+      end
+
+      context "exists past model" do
+        before do
+          EmployeeWithUniquness.create!(name: "Tom", created_at: "1982/12/02", deleted_at: "2001/03/24")
+        end
+        it { is_expected.not_to raise_error }
+      end
+
+      context "exists future model" do
+        before do
+          EmployeeWithUniquness.create!(name: "Tom", created_at: Time.current + 10.days)
+        end
+        it { is_expected.to raise_error ActiveRecord::RecordInvalid }
+      end
+    end
+
+    context "and `valid_from`" do
+      before do
+        EmployeeWithUniquness.create(name: "Jane", valid_from: "2019/1/10", valid_to: "2019/20")
```

Comment:
`2019/20` 🤔

Reply:
Thank!
Remove it and fixed other test. https://github.com/kufu/activerecord-bitemporal/pull/33/commits/238b642c6df62b01ea7b86f6c741cce976d99d8f